src/libs/antares/study/parts/hydro/datatransfer.cpp

```diff
@@ -0,0 +1,171 @@
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "../../study.h"
+#include "datatransfer.h"
+#include <antares/inifile/inifile.h>
+
+using namespace Yuni;
+
+#define SEP IO::Separator
+
+namespace Antares::Data
+{
+
+DataTransfer::DataTransfer()
+{
+    maxPower.reset(4, DAYS_PER_YEAR, true);
+}
+
+bool DataTransfer::LoadFromFolder(Study& study, const AnyString& folder, Area& area)
+{
+    auto& buffer = study.bufferLoadingTS;
+    bool ret = true;
+
+    buffer.clear() << folder << SEP << "common" << SEP << "capacity" << SEP << "maxpower_"
+                   << area.id << '.' << study.inputExtension;
+
+    // GUI part patch :
+    // We need to know, when estimating the RAM required by the solver, if the current area
+    // is hydro modulable. Therefore, reading the area's daily max power at this stage is
+    // necessary.
+
+    if (!study.usedByTheSolver)
+    {
+        bool enabledModeIsChanged = false;
+        if (JIT::enabled)
+        {
+            JIT::enabled = false; // Allowing to read the area's daily max power
+            enabledModeIsChanged = true;
+        }
+
+        ret = maxPower.loadFromCSVFile(
+                buffer, 4, DAYS_PER_YEAR, Matrix<>::optFixedSize, &study.dataBuffer)
+              && ret;
+
+        if (enabledModeIsChanged)
+            JIT::enabled = true; // Back to the previous loading mode.
+    }
+    else
+    {
+        ret = maxPower.loadFromCSVFile(
+                buffer, 4, DAYS_PER_YEAR, Matrix<>::optFixedSize, &study.dataBuffer)
+              && ret;
+
+        bool errorPowers = false;
+        for (uint i = 0; i < 4; ++i)
+        {
+            auto& col = maxPower[i];
+            for (uint day = 0; day < DAYS_PER_YEAR; ++day)
+            {
+                if (!errorPowers && (col[day] < 0 || (i % 2 /*column hours*/ && col[day] > 24)))
+                {
+                    logs.error() << area.name << ": invalid power or energy value";
+                    errorPowers = true;
+                    ret = false;
+                }
+            }
+        }
+    }
+
+    return ret;
+}
+
+bool DataTransfer::AutoTransferHours(Study& study, const AnyString& folder, Area& area)
+{
+    auto& buffer = study.bufferLoadingTS;
+    bool ret = true;
+
+    auto& maxHoursGen = area.hydro.maxHoursGen;
```

Comment:
I put it as maxHours because those data are in fact hours in one day which are used to calculate daily energy by multiplying with mean daily maximum generation/pumping power. I don't think we should go with maxHourly, instead we should go with maxDaily name convention.